src/main/java/io/vertx/core/logging/Log4jLogDelegate.java

```diff
@@ -101,4 +150,8 @@ private void log(Level level, Object message) {
   private void log(Level level, Object message, Throwable t) {
     logger.log(FQCN, level, message, t);
   }
+
+  private void throwUnsupportedOperationException() {
```

Comment:
We actually started off down the route of computing our own formatted messages but decided to delegate to the logging library as most probably the logging library implementation is much better optimised.

I think it's probably ok for the new methods on the Log4JLogDelegate to throw UnsupportedOperationException as this won't break existing code as the methods are new.